crates/telio-utils/src/local_interfaces.rs

```diff
@@ -0,0 +1,93 @@
+use ipnet::Ipv4Net;
+use std::net::{IpAddr, Ipv4Addr};
+
+/// Function type of a function returning interface addresses from the system
+pub type GetIfAddrs = fn() -> std::io::Result<Vec<if_addrs::Interface>>;
+
+/// Function returning interface addresses from the system
+pub fn system_get_if_addr() -> std::io::Result<Vec<if_addrs::Interface>> {
+    if_addrs::get_if_addrs()
+}
+
+/// Method that returns vector of Interfaces on the system
+/// Filtering out meshnet and loopback IP
+pub fn gather_local_interfaces(
```

Comment:
IMO this should be part of the `telio-network-monitor` crate, no need to have it isolated here in `telio-utils`

Reply:
eg: `NetworkMonitor::gather_local_interfaces()`

Reply:
Yep. That makes better sense!